### ISSUE TITLE
end ussd session

### ISSUE THREAD

**Issue by adamjiu** [wizicer/UAPNetwork#10]
Is there a way i can end the session in UAPTester sample program?

**Comment by wizicer** [comment]
Until it reads `end`:
https://github.com/wizicer/UAPNetwork/blob/master/UAPTester/Program.cs#L48

**Comment by adamjiu** [comment]
Ok thanks.If i add a menu as in Example  Menu below:
1. Buy Airtime
2. Buy Bundles

After Buy Airtime it goes to a submenu below:
1.Buy 1 dollar airtime
2.Buy 10 dollar airtime.

After someone selects the submenu "1.Buy 1 dollar airtime" , How can i get the UssdContent? it returns 1 in this case yet it should be 1*1 because its a submenu under a menu

**Comment by wizicer** [comment]
You should have a session to store user's current status.

**Comment by adamjiu** [comment]
i thought the sessions are handled automatically, i tried with 2 numbers and each one was captured differently. The problem is Submenus. There are not passing the UssdContent of the Main menu so that i can capture the Main Menu e.g 1 and Submenu e.g 1. Please help with sample code

**Comment by wizicer** [comment]
No, this library is only a Ussd protocol communicator, session manager is not including, you need to build yourself.

**Comment by adamjiu** [comment]
ok thanks. Just for some guideline: UapBegin is for starting the USSD and UapEnd is for ending it, right?

**Comment by rontuku** [comment]
Hi adamjiu,
To end the session here is a sample code

                    UapContinue con = new UapContinue()
                    {
                        SenderId = seq++,
                        ReceiverId = e.BeginUap.SenderId,
                        UssdVersion = e.BeginUap.UssdVersion,
                        UssdOpType = UapBase.UssdOpTypeEnum.Response,
                        MsIsdn = e.BeginUap.MsIsdn,
                        ServiceCode = e.BeginUap.ServiceCode,
                        CodeScheme = 0x08,
                        UssdContent = Response,
                        CommandId = UapBase.CommandIdType.End
                    };
                    ((UapCommunicator)source).SendUap(con);

note this part  **CommandId = UapBase.CommandIdType.End**
But as wizicer mentioned you have to manage the user session separately.

**Comment by adamjiu** [comment]
thanks a lot Ronald.  I am doing this in a console application which doesn't support sessions. Please advice how to handle the session, will appreciate a lot if you assist with some guideline or code
